flutter-idea/src/io/flutter/analytics/UnifiedAnalytics.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2024 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.analytics;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.jetbrains.lang.dart.ide.toolingDaemon.DartToolingDaemonConsumer;
+import com.jetbrains.lang.dart.ide.toolingDaemon.DartToolingDaemonService;
+import de.roderick.weberknecht.WebSocketException;
+import io.flutter.sdk.FlutterSdkUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Facilitates sending information to unified analytics.
+ */
+public class UnifiedAnalytics {
+  private static final Logger LOG = Logger.getInstance(UnifiedAnalytics.class);
+
+  @Nullable Boolean enabled = null;
+  final Project project;
+  final DartToolingDaemonService dtdService;
+  @NotNull final FlutterSdkUtil flutterSdkUtil;
+
+
+  public UnifiedAnalytics(@NotNull Project project) {
+    this.project = project;
+    this.dtdService = DartToolingDaemonService.getInstance(project);
+    this.flutterSdkUtil = new FlutterSdkUtil();
+  }
+
+  public void manageConsent() {
```

Comment:
Does this code play well with the existing analysis server integration with unified_analytics?

Reply:
It sounds like once IntelliJ is marked as consented in the analytics file, then analysis server will be able to use this? I'll have to check with @bwilkerson whether this works once we turn this on, but that will have to wait until after I return.